numba/core/ir.py

```diff
@@ -138,7 +138,7 @@ def strformat(self, nlines_up=2):
 
 
         ret = [] # accumulates output
-        if lines and use_line:
+        if lines and use_line > 0:
```

Comment:
I think e0c54bde from #8621 "fixes" the root cause of the problem (because it also hits DWARF output, can't have negative line numbers) but putting in this protection in the user facing part is probably a good idea anyway.